### PR TITLE
Feature 3 mark plant as owned

### PR DESCRIPTION
**Pull request by StephMode**
This PR is part of the work to build the MVP of the app and lay the foundation for further improvements. 
This feature enbales users to mark plants from the HomePage as owned and easily manage their collection on a dedicated page. 

To achieve this, we:

- Created a PlantOwnedButton component that lets users mark plants as owned or unowned with a simple toggle.
- Set up persistent storage using localStorage to ensure that users' owned plants remain saved across sessions.
- Developed a new “My Plants” page that displays only the plants marked as owned, with a message displayed when no plants are owned.
- Added navigation links between the main plant list and the “My Plants” page for easy access.

### CODE REVIEW

File: components/PlantOwnedButton/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const StyledButton = styled.button`
+  border: none;
+  background: var(--green-main);
+  border-radius: 100%;
+  position: absolute;
+  top: 10px;
+  right: 10px;
+  padding: 10px;
+  fill: blue;
```
Comment (benjaminpfleghaar) [inline components/PlantOwnedButton/index.js:33]: The fill attribute only applies to svg related elements but not on buttons:
https://developer.mozilla.org/en-US/docs/Web/CSS/fill

File: pages/_app.js [hunk 2]
```diff
@@ -1,28 +1,37 @@
...
+    defaultValue: plants,
+  });
+
+  function handleToggleOwned(id) {
+    setAllPlants((prevPlants) =>
+      prevPlants.map((plant) =>
+        plant.id === id ? { ...plant, isOwned: !plant.isOwned } : plant
+      )
+    );
+  }
```
Comment (benjaminpfleghaar) [inline pages/_app.js:17]: Great useState updater function 👍

File: components/PlantOwnedButton/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+export default function PlantOwnedButton({
+  isOwned,
+  handleToggleOwned,
+  plantId,
+}) {
+  return (
+    <StyledButton type="button" onClick={() => handleToggleOwned(plantId)}>
+      {isOwned === true ? (
+        <Image
+          src={"./heart-solid.svg"}
```
Comment (benjaminpfleghaar) [inline components/PlantOwnedButton/index.js:13]: Short hint: No need to use { } for the src attribute as long as you write plain text.

**Review by benjaminpfleghaar — COMMENTED** [review]
Great work, added just a few comments for improving your code.

File: components/PlantOwnedButton/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          height={40}
+          width={40}
+          alt="heart icon"
+        />
+      ) : (
+        <Image src={"./heart.svg"} height={40} width={40} alt="heart icon" />
+      )}
+    </StyledButton>
```
[the remark this answers is not in the document]
Comment (levinthiel) [reply components/PlantOwnedButton/index.js:21]: Thx for the great idea

File: styles.js [hunk 5]
```diff
@@ -25,12 +28,32 @@ export default createGlobalStyle`
...
     font-family: system-ui;
     width: 100%;
   }
+  header {
+    display: flex;
+    padding: 5px 20px;
+    width: 100%;
+    height: 50px;
+    background-color: var(--green-main);
+  }
```
[the remark this answers is not in the document]
Comment (levinthiel) [reply styles.js:42]: true, thx

File: pages/_app.js [hunk 6]
```diff
@@ -1,28 +1,37 @@
...
 import Image from "next/image";
-import styled from "styled-components";
+import plants from "/lib/data";
+import useLocalStorageState from "use-local-storage-state";
 
 export default function App({ Component, pageProps }) {
-  
+  const [allPlants, setAllPlants] = useLocalStorageState("allPlants", {
+    defaultValue: plants,
+  });
```
Comment (JessicaLoers) [inline pages/_app.js:9]: **As mentioned above** ⬆️

```suggestion
const [plants, setPlants] = useLocalStorageState("plants", {
  defaultValue: initialPlants,
});

```

File: pages/_app.js [hunk 7]
```diff
@@ -1,28 +1,37 @@
...
+          height={50}
           alt={"rooted logo"}
         />
-      </StyledHeader>
-      <Component {...pageProps} />
+      </header>
+      <Component
+        {...pageProps}
+        handleToggleOwned={handleToggleOwned}
+        plants={allPlants}
```
Comment (JessicaLoers) [inline pages/_app.js:33]: As mentioned above ⬆️
```suggestion
        plants={plants}
```

File: pages/_app.js [hunk 8]
```diff
@@ -1,28 +1,37 @@
...
-      <StyledHeader>
-        <Image 
+      <header>
+        <Image
           src={"/logo-main.svg"}
-          width={150}
-          height={30}
+          width={200}
+          height={50}
           alt={"rooted logo"}
```
Comment (JessicaLoers) [inline pages/_app.js:27]: As mentioned above...
```suggestion
          alt="rooted logo"
```

File: pages/_app.js [hunk 9]
```diff
@@ -1,28 +1,37 @@
...
+  }
+
   return (
     <>
       <GlobalStyle />
-      <StyledHeader>
-        <Image 
+      <header>
+        <Image
           src={"/logo-main.svg"}
```
Comment (JessicaLoers) [inline pages/_app.js:24]: _Just a tiny note:_ The curly braces `{}` are not needed for plain strings in JSX attributes.

```suggestion
          src="/logo-main.svg"
```

File: pages/index.js [hunk 10]
```diff
@@ -1,21 +1,25 @@
...
+              image={plant.imageUrl}
+              name={plant.name}
+              botanicalName={plant.botanicalName}
+              handleToggleOwned={handleToggleOwned}
+              isOwned={plant.isOwned}
+            />
+          </li>
+        ))}
+      </ul>
+      <Link href={"/myplants"}>My Plants</Link>
```
Comment (JessicaLoers) [inline pages/index.js:22]: `Tiny note: `Curly braces `{}` are unnecessary for plain strings
```suggestion
      <Link href="/myplants">My Plants</Link>
```

File: pages/myplants.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+              image={plant.imageUrl}
+              name={plant.name}
+              botanicalName={plant.botanicalName}
+              handleToggleOwned={handleToggleOwned}
+              isOwned={plant.isOwned}
+            />
+          </li>
+        ))}
+      </ul>
+      <Link href={"/"}>Plant List</Link>
```
Comment (JessicaLoers) [inline pages/myplants.js:28]:
```suggestion
      <Link href="/">Plant List</Link>
```

File: pages/_app.js [hunk 12]
```diff
@@ -1,28 +1,37 @@
...
+      )
+    );
+  }
+
   return (
     <>
       <GlobalStyle />
-      <StyledHeader>
-        <Image 
+      <header>
```
Comment (JessicaLoers) [inline pages/_app.js:22]: Consider using a [Layout component pattern](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts#layout-pattern) from Next.js to wrap the main structure of the page, including `<header>`, <`main>` and later the possible `<footer>` .

This approach can help with maintaining consistent styling 
and structure across different pages, making it easier to apply shared layout elements. It also improves code organization, as the layout logic is centralized.

**But that can be a task for the next print. 🏃**

File: components/PlantOwnedButton/index.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    <StyledButton type="button" onClick={() => handleToggleOwned(plantId)}>
+      <Image
+        src={isOwned ? "./heart-solid.svg" : "./heart.svg"}
+        height={40}
+        width={40}
+        alt="heart icon"
+      />
+    </StyledButton>
+  );
+}
```
[the remark this answers is not in the document]
Comment (levinthiel) [reply components/PlantOwnedButton/index.js:19]: Thanks, that's a little tricky to wrap our head around, but we implemented it and it works perfectly <3

File: pages/_app.js (hunk 7 above)
Comment (levinthiel) [reply pages/_app.js:33]: Thanks, it makes indeed more sense like this

**Pull request merged**